### PR TITLE
Use `folder-hash` instead of `hash-files`.

### LINKED ISSUE #44: Node `Buffer`-related deprecation warning
```
(node:1636) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
```
Seen in https://github.com/tokio-rs/tokio/actions/runs/4788275544/jobs/8514678819

Treating this as a bug.

### PR DESCRIPTION
**Pull request by mgr0dzicki**
Closes #44.

The deprecation warning originated in package `hash-files`. It was the simplest one to use, but hasn't been maintained since 2016. I switched to `folder-hash` (https://www.npmjs.com/package/folder-hash), with latest release from January, which seems to be the most popular one now.

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+    clearMocks: true,
+    moduleFileExtensions: ['js', 'ts'],
+    testMatch: ['**/*.test.ts'],
+    transform: {
+      '^.+\\.ts$': 'ts-jest'
+    },
+    verbose: true
+  }
```
Comment (obi1kenobi) [inline jest.config.js:9]: Something weird happened with the indentation here, could you take a look? This is all in a mix of 2-spaced and 4-spaced indentation, and the last line is unnecessarily indented.

File: src/utils.ts [hunk 2]
```diff
@@ -34,12 +36,15 @@ export function optionFromList(option: string, values: string[]): string[] {
...
-        return "";
-    }
+export async function hashFolderContent(path: string): Promise<string> {
+    return (await hashElement(path, { encoding: "hex", folders: { ignoreRootName: true } })).hash;
+}
+
+export async function hashFiles(patterns: string[] = []): Promise<string> {
+    const hasher = crypto.createHash("md5");
+    const nodes = (await glob(patterns)).sort().map((filename) => hashElement(filename));
+    (await Promise.all(nodes)).forEach((node) => hasher.update(node.hash));
```
Comment (obi1kenobi) [inline src/utils.ts:46]: In both of these cases, can you assign the `await` result to an intermediate variable and then use it? `(await <complex expression>).<stuff>` is not very readable, and `return (await <complex expr>).<stuff>` is even worse.

**Review by obi1kenobi — COMMENTED** [review]
Love the fix, just some stylistic concerns.

File: src/utils.ts (hunk 2 above)
Comment (mgr0dzicki) [reply src/utils.ts]: You're right, I hadn't noticed how ugly it was before. I've changed it, could you have a look if it is better now? ;)

Comment (obi1kenobi) [reply src/utils.ts]: Sure, will look in a minute!

For situations like this, self-reviewing your PR in the GitHub interface right after you open it is super useful. When you're in the middle of trying to get things working, things like these are easy to miss. But I bet you would have noticed both issues I flagged here by yourself in a self-review — I don't think I contributed anything special there.

**Review by obi1kenobi — APPROVED** [review]

**Comment by obi1kenobi** [comment]
Would you care to tag the latest `main` as `v2.1` and then move the `v2` tag to that new commit as well?

Remember that it's important to minimize the time the repo is missing a `v2` tag, because any actions that use the tag and run in that time will fail.

For best results, I recommend:
- make the `v2.1` tag locally and push it
- delete the `v2` tag locally
- make the new `v2 tag locally
- `git push origin <tag-name> --delete && git push origin <tag-name>` to replace the v2 tag as quickly as possible

I trust you to do this. If for any reason you'd prefer not to, just let me know and I'll take care of it.

File: src/utils.ts (hunk 2 above)
Comment (mgr0dzicki) [reply src/utils.ts]: I always go through the changes I made in the GitHub interface, and this was the case too, but maybe I wasn't careful enough. I'll try to do better next time!

**Comment by mgr0dzicki** [comment]
Done! Thank you for the detailed instruction, it was very useful ;)

**Comment by obi1kenobi** [comment]
Happy to help!

**Pull request merged**